shared/src/main/scala/fr/hmil/roshttp/SimpleHttpResponse.scala

```diff
@@ -0,0 +1,59 @@
+package fr.hmil.roshttp
+
+import java.nio.ByteBuffer
+import java.util.concurrent.TimeUnit
+
+import monifu.reactive.Observable
+
+import monifu.concurrent.Implicits.globalScheduler
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * An HTTP response obtained via an [[HttpRequest]]
+ */
+class SimpleHttpResponse(
+    val statusCode: Int,
+    val body: String,
+    val headers: HeaderMap[String])
+  extends HttpResponse
+
+object SimpleHttpResponse extends HttpResponseFactory[SimpleHttpResponse] {
+  override def apply(
+      statusCode: Int,
+      bodyStream: Observable[ByteBuffer],
+      headers: HeaderMap[String])
+      (implicit ec: ExecutionContext, config: HttpConfig): Future[SimpleHttpResponse] = {
+
+    val charset = HttpUtils.charsetFromContentType(headers.getOrElse("content-type", null))
+
+    bodyStream
+      .buffer(FiniteDuration(config.bodyCollectTimeout, TimeUnit.SECONDS))
+      .map({ seq =>
+        // Allocate maximum expected body length
+        val buffer = ByteBuffer.allocate(seq.length * config.streamChunkSize)
+        val totalBytes = seq.foldLeft(0)({(count, chunk) =>
+          buffer.put(chunk)
+          count + chunk.limit
+        })
+        buffer.limit(totalBytes)
+        getStringFromBuffer(buffer, charset)
+      })
+      // TODO: We might not want to depend on monifu's Scheduler
+      .asFuture
```

Comment:
> I should then probably replace all ExecutionContext parameters with Scheduler instead to make the whole API more consistent.

I don't think so. I think it's good to require as little as possible from your users. That way, say they're only using the batch APIs, they can use any EC they already have. If they're using the streaming API as well, they can have an implicit `Scheduler`, and it will *also* be valid for methods requiring an implicit EC because `Scheduler <: EC`.